app/src/main/java/com/example/maintenancespace/ui/cars/CarFragment.java

```diff
@@ -44,10 +67,10 @@ public void onCarFetched(CarModel car) {
             public void onCarsFetched(ArrayList<CarModel> cars) {
                 final FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
                 for (CarModel car : cars) {
-                    CarListItemFragment existingFragment = (CarListItemFragment) fragmentManager.findFragmentByTag(car.getVin());
+                    CarListItemFragment existingFragment = (CarListItemFragment) fragmentManager.findFragmentByTag(car.getId());
                     if (existingFragment == null) {
                         CarListItemFragment eventFragment = CarListItemFragment.newInstance(car);
-                        fragmentTransaction.add(R.id.car_layout, eventFragment, car.getVin());
+                        fragmentTransaction.add(R.id.car_layout, eventFragment, car.getId());
```

Comment:
Updated this to use the car document id instead since this will always be unique no matter the user input.